pkg/cmd/pr/checks/checks.go

```diff
@@ -131,33 +130,39 @@ func checksRun(opts *ChecksOptions) error {
 	var checks []check
 	var counts checkCounts
 
+	if opts.Watch {
+		opts.IO.StartAlternateScreenBuffer()
+	}
+
+	// Do not return err until we can StopAlternateScreenBuffer()
```

Comment:
Would it make more sense to use `defer StopAlternateScreenBuffer()` here instead of `break`s?

Reply:
That's actually what I did originally, but incorrectly remembered that a `defer` happens at the end of the calling `func` - not just the scope e.g., conditional block. What ends up happening then is that anything else that gets printed gets printed to the ASB and later lost when the ASB is destroyed.

The same problem happens for the progress indicator, which I basically pilfered for my own extensions and thought I was clever using `defer`: in that case, anything else that gets printed throws off the row and/or column to which the progress indicator was printed and the output buffer gets all wonky i.e., the last line remains in the output buffer.

As a result, I considered refactoring part of this into its own `func` but it ended up looking a bit much e.g., more maintenance cost, just for the purpose of using `defer`. All things considered, this seemed cleaner but I'm open to alternate suggestions.

Reply:
got it!

I think it's ok the way it is -- it's certainly readable and understandable. Thanks for the thorough explanation :)